c++/src/kj/compat/http.c++

```diff
@@ -5049,6 +5762,49 @@ private:
       }
 
       KJ_SWITCH_ONEOF(requestOrProtocolError) {
+        KJ_CASE_ONEOF(request, HttpHeaders::ConnectRequest) {
+          auto& headers = httpInput.getHeaders();
+
+          currentMethod = HttpConnectMethod();
+
+          SuspendableRequest suspendable(*this, HttpConnectMethod(), request.authority, headers);
+          auto maybeService = factory(suspendable);
+
+          if (suspended) {
+            return false;
+          }
+
+          auto service = KJ_ASSERT_NONNULL(kj::mv(maybeService),
+              "SuspendableHttpServiceFactory did not suspend, but returned nullptr.");
+
+          // TODO(perf): If the client disconnects, should we cancel the response? Probably, to
```

Comment:
It looks like you copied this comment from the non-connect path. I think the comment is actually obsolete -- we actually cancel out the whole loop if `whenWriteDisconnected()` resolves.